src/components/CodeInput.tsx

```diff
@@ -30,81 +10,42 @@ export interface CodeInputProps {
 }
 
 export const CodeInput = ({value, onChange, accessibilityLabel}: CodeInputProps) => {
-  const inputRef = useRef<TextInput>(null);
   const onChangeTrimmed = useCallback(text => onChange(text.trim()), [onChange]);
-
-  const displayValue = useCallback(
-    () =>
-      Array(8)
-        .fill(null)
-        .map((_, x) => {
-          const characterToDisplay = value[x] || ' ';
-          return (
-            <Box /* eslint-disable-next-line react/no-array-index-key */
-              key={`input-${x}`}
-              flex={1}
-              marginHorizontal="none"
-              marginRight={x === 3 ? 'm' : 'none'}
-              borderRadius={9}
-              borderWidth={4}
-              borderColor={inputFocusColor(value, x)}
-            >
-              <Box
-                flex={1}
-                paddingHorizontal="xs"
-                borderWidth={2}
-                borderColor={inputBorderColor(value, x)}
-                borderRadius={5}
-              >
-                <Text variant="codeInput" color="overlayBodyText" textAlign="center" paddingVertical="xs">
-                  {characterToDisplay}
-                </Text>
-              </Box>
-            </Box>
-          );
-        }),
-    [value],
-  );
-
-  const giveFocus = useCallback(() => inputRef.current?.focus(), []);
+  const [isFocus, setIsFocus] = useState(false);
+  const onFocus = useCallback(() => setIsFocus(true), []);
+  const onBlur = useCallback(() => setIsFocus(false), []);
 
   return (
     <>
-      <TextInput
-        value={value}
-        ref={inputRef}
-        onChangeText={onChangeTrimmed}
-        keyboardType="number-pad"
-        autoCorrect={false}
-        autoCompleteType="off"
-        returnKeyType="done"
-        accessibilityLabel={accessibilityLabel}
-        caretHidden
-        maxLength={8}
-        style={styles.input}
-      />
-      <TouchableWithoutFeedback
-        onPress={giveFocus}
-        accessibilityElementsHidden
-        importantForAccessibility="no-hide-descendants"
+      <Box
+        marginHorizontal="none"
+        borderRadius={9}
+        borderWidth={4}
+        borderColor={isFocus ? 'focus' : 'overlayBackground'}
       >
-        <Box flexDirection="row" justifyContent="space-evenly" marginHorizontal="m" paddingVertical="l">
-          {displayValue()}
+        <Box
+          flex={1}
+          paddingHorizontal="xs"
+          borderWidth={2}
+          borderColor={isFocus ? 'overlayBodyText' : 'gray2'}
+          borderRadius={5}
+        >
+          <TextInput
+            value={value}
+            onChangeText={onChangeTrimmed}
+            onFocus={onFocus}
+            onBlur={onBlur}
+            autoCorrect={false}
+            autoCompleteType="off"
+            returnKeyType="done"
+            accessibilityLabel={accessibilityLabel}
+            padding="s"
+            maxLength={8}
+            fontSize={26}
+            borderWidth={0}
```

Comment:
I was playing around and if I can suggest these attributes I think it would be cool!
```
autoCapitalize="characters"
fontFamily="Menlo"
letterSpacing={5}
```

Reply:
Added! LGTM